core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -170,8 +169,9 @@ abstract class AbstractFetcherThread(name: String,
           val topic = topicPartition.topic
           val partitionId = topicPartition.partition
           Option(partitionStates.stateValue(topicPartition)).foreach(currentPartitionFetchState =>
-            // we append to the log if the current offset is defined and it is the same as the offset requested during fetch
-            if (fetchRequest.offset(topicPartition) == currentPartitionFetchState.fetchOffset) {
+            // we append to the log if the current offset is the same as the offset requested and if the partition state is ready for fetch
+            if (fetchRequest.offset(topicPartition) == currentPartitionFetchState.fetchOffset &&
```

Comment:
How about we change the comment to the following?

"It's possible that a partition is removed and re-added or truncated when there is a pending fetch request. In this case, we only want to process the fetch response if the partition state is ready for fetch and the current offset is the same as the offset requested."

Reply:
Sure. I have updated the comment as suggested.